index.d.ts

```diff
@@ -0,0 +1,116 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import { Http2ServerRequest, Http2ServerResponse } from 'http2';
+
+declare function Router<V extends Router.HTTPVersion = Router.HTTPVersion.V1>(
+  config?: Router.Config<V>
+): Router.Instance<V>;
+
+declare namespace Router {
+  enum HTTPVersion {
+    V1 = 'http1',
+    V2 = 'http2'
+  }
+
+  type HTTPMethod =
+    | 'DELETE'
+    | 'GET'
+    | 'HEAD'
+    | 'PATCH'
+    | 'POST'
+    | 'PUT'
+    | 'OPTIONS';
+
+  type Handler<V extends HTTPVersion> = (
+    req: V extends HTTPVersion.V1 ? IncomingMessage : Http2ServerRequest,
+    res: V extends HTTPVersion.V1 ? ServerResponse : Http2ServerResponse,
+    params: { [k: string]: string | undefined },
+    store: any
```

Comment:
I guess `params` and `store` can be optional.

Reply:
I noticed when using the library in a node repl that you always return params. It's just an empty object if there are no params in matched route. As for store, since its opaque, I had to use any which trumps optional/nullable (`any | null === any`, `any | undefined === any`, `store?: any === store: any`)

Reply:
Note the value of params for different cases in the screenshot below:
![image](https://user-images.githubusercontent.com/678548/49537805-df16f380-f8c1-11e8-92bd-9bb15b6f604c.png)

Reply:
Yes, I'm always returning them, but what I mean is that you don't have to declare them always, with this declaration you have to, right?
```js
router.on('GET', '/', (req, res) => {}) // this is perfectly fine
```

Reply:
Good to know, thanks!